registration/models.py

```diff
@@ -76,8 +76,9 @@ class BetaTestApplication(ValidateModelMixin, TimeStampedModel):
                 'The subdomain name must at least have 3 characters.',
             ),
             validators.RegexValidator(
-                r'^[\w.-]+$',
-                "Please include only letters, numbers, '_', '-' and '.'",
+                r'^[a-zA-Z0-9](?:[a-zA-Z0-9\-]{1,63}[a-zA-Z0-9])$',
```

Comment:
If I remember correctly, the maximum length of a domain name is 63 characters, but this regex matches names of up to 65 characters in length.

I suggest enforcing the length via the `max_length` property of the field rather than the regex.  This would allow us to simplify the regex to 

    r'^[a-zA-Z0-9]([a-zA-Z0-9\-]+[a-zA-Z0-9])?$'

This would also make the `MinLengthValidator` redundant, but I'd keep it for clarity (and in case we want to increase the minimum length).

Reply:
Ah, that sounds good. I'll address this when we pick the ticket back up.